lib/puppet/type/ec2_elastic_ip.rb

```diff
@@ -35,6 +35,14 @@
     end
   end
 
+  newproperty(:instance_id) do
+    desc 'The id of the instance associated with the Elastic IP'
+    validate do |value|
+      fail 'instance_id should be a String' unless value.is_a?(String)
+      fail 'You must provide an instance_id for the Elastic IP association' if value.nil? || value.empty?
```

Comment:
I don't think this is the correct approach, or at least isn't in keeping with the module's current design. Using the ID in extremis maybe, but requiring it as the primary interface I'm less sure about I'm afraid.